rskj-core/src/test/java/org/ethereum/net/p2p/PeersMessageTest.java

```diff
@@ -0,0 +1,101 @@
+package org.ethereum.net.p2p;
+
+
+import co.rsk.util.RLPException;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.TestUtils;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PeersMessageTest {
+    //This is the result of enconding the test peerList
+    private static String encodedHexString = "db05cc847649472482b6de821010c0cc8473cce52482f8ab821010c0";
+
+    @Test
+    void peerMessageIsCreatedProperly() throws UnknownHostException {
+        byte[] payload = getTestPeerMessagePayload();
+        PeersMessage message = new PeersMessage(payload);
+        Set<PeerConnectionData> peerList = message.getPeers();
+        List<PeerConnectionData> expectedPeerList = getTestPeerConnectionDataList();
+
+        assertTrue(expectedPeerList.size() == peerList.size() && peerList.containsAll(expectedPeerList));
+    }
+
+    @Test
+    void getPeersFromCreatePeers() throws UnknownHostException {
+        List<PeerConnectionData> peerList = getTestPeerConnectionDataList();
+        Set<PeerConnectionData> peerSet = new HashSet<>(peerList);
+        PeersMessage peersMessage = new PeersMessage(peerSet);
+
+        Set<PeerConnectionData> messagePeers = peersMessage.getPeers();
+        assertEquals(messagePeers, peerSet);
+        byte[] encodedData = peersMessage.getEncoded();
+        assertNotNull(encodedData);
+    }
+
+    @Test
+    void parseFunctionIsCalledWith() {
+        byte[] payload = TestUtils.generateBytes(PeersMessage.class
+                , "payload", 1);
+        PeersMessage message = new PeersMessage(payload);
+        byte[] encoded = message.getEncoded();
+        assertEquals(payload, encoded, "Payload must be used as encoded data");
+        assertThrows(RLPException.class, () -> message.getPeers(), "Exception expected as payload is not valid and parse method is called");
+    }
+
+    @Test
+    void encodeFunctionIsCalledProperly() {
+        Set<PeerConnectionData> set = new HashSet<>(Arrays.asList(Mockito.mock(PeerConnectionData.class)));
+        PeersMessage peersMessage = new PeersMessage(set);
+
+        assertNotNull(peersMessage.getPeers(), "Parse Method is not called");
```

Comment:
WDYT? A bit more clear:

```suggestion
        assertNotNull(peersMessage.getPeers(), "parse() expected but not called");
```